tests/ci/integration/run_ruby_integration.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash -exu
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+source tests/ci/common_posix_setup.sh
+
+set -exuo pipefail
+
+# Set up environment.
+
+# SYS_ROOT
+#  - SRC_ROOT(aws-lc)
+#    - SCRATCH_FOLDER
+#      - RUBY_SRC_FOLDER
+#        - ruby_3_1
+#      - RUBY_PATCH_FOLDER
+#        - ruby_3_1
+#      - AWS_LC_BUILD_FOLDER
+#      - AWS_LC_INSTALL_FOLDER
+
+# Assumes script is executed from the root of aws-lc directory
+SCRATCH_FOLDER="${SRC_ROOT}/RUBY_BUILD_ROOT"
+RUBY_SRC_FOLDER="${SCRATCH_FOLDER}/ruby-src"
+RUBY_PATCH_FOLDER="${SRC_ROOT}/tests/ci/integration/ruby_patch"
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${SCRATCH_FOLDER}/aws-lc-install"
+
+function ruby_build() {
+    local branch=${1}
+    pushd ${branch}
+    ./autogen.sh
+    mkdir -p install
+    ./configure --disable-install-doc \
+                 --prefix="$PWD/install" \
+                 --with-openssl-dir=${AWS_LC_INSTALL_FOLDER}
+    make V=1 -j ${NUM_CPU_THREADS}
+    make install
+    # Check that AWS-LC was used.
+    ./install/bin/ruby -e 'require "openssl"; puts OpenSSL::OPENSSL_VERSION' | grep -q "AWS-LC" && echo "AWS-LC found!" || exit 1
+
+    #TODO: add more relevant tests here
+    make test-all TESTS="test/openssl/*.rb"
```

Comment:
I may be misunderstanding this, but from this PR's most recent CI run, it [looks like](https://github.com/aws/aws-lc/actions/runs/12248575984/job/34168439237?pr=1563#step:4:4549) a different executable than `./install/bin/ruby` is used to execute the tests, namely `miniruby`:

```

test-all
  Run options: 
    --seed=24196
    "--ruby=./miniruby -I./lib -I. -I.ext/common  ./tool/runruby.rb --extout=.ext  -- --disable-gems"
    --excludes-dir=./test/excludes
    --name=!/memory_leak/
```

could we patch a test into the OpenSSL module unit tests that asserts `OpenSSL::OPENSSL_VERSION.include? 'AWS-LC'`?

Reply:
Not a test, but we do have a version detection mechanism in the patch:
```
+ def aws_lc?(major = nil, minor = nil, fix = nil)
+    version = OpenSSL::OPENSSL_VERSION.scan(/AWS-LC (\d+)\.(\d+)\.(\d+).*/)[0]
+    return false unless version
+    !major || (version.map(&:to_i) <=> [major, minor, fix]) >= 0
+  end
```
It's used throughout the tests to skip across various AWS-LC specific discrepancies. I'm hesitant to add a specific test that asserts AWS-LC. Ultimate end goal is to run this patch down to 0 if upstream is willing to take us. I also think the various bash checks we do and the mentioned version detection should be sufficient.

Reply:
`miniruby` appears to be minimal lighter version of ruby used in their build. The ruby binary we're asserting in this script is the actual ruby binary that the general consumer uses.
* https://github.com/ruby/ruby/wiki/Developer-How-To#make-targets

Reply:
> miniruby appears to be minimal lighter version of ruby used in their build

right. i think we need some way to assert that the `miniruby` used to run the tests is actually linked against AWS-LC. otherwise, we might be testing against an OpenSSL-backed `miniruby`, which is meaningless to us.

Reply:
Added an assertion for miniruby